src/rpm_ostree/cli_status.rs

```diff
@@ -85,37 +85,62 @@ impl Deployment {
             checksum: self.base_revision(),
             version: self.version,
             age_index: None,
+            is_oci: self.container_image_reference.is_some(),
         }
     }
 
     /// Return the deployment base revision.
     pub fn base_revision(&self) -> String {
-        self.base_checksum
+        self.container_image_reference
             .clone()
+            .map(|pullspec| {
+                pullspec
+                    .strip_prefix("ostree-remote-image:fedora:docker://")
+                    .map(|s| s.to_string())
+            })
+            .flatten()
+            .or(self.base_checksum.clone())
             .unwrap_or_else(|| self.checksum.clone())
     }
 }
 
 /// Parse the booted deployment from status object.
 pub fn parse_booted(status: &Status) -> Result<Release> {
     let status = booted_status(status)?;
-    if let Some(img) = status.container_image_reference.as_ref() {
-        let msg = format!("Automatic updates disabled; booted into container image {img}");
-        crate::utils::notify_ready();
-        crate::utils::update_unit_status(&msg);
-        return Err(anyhow::Error::new(SystemInoperable(msg)));
-    }
     Ok(status.into_release())
 }
 
 fn fedora_coreos_stream_from_deployment(deploy: &Deployment) -> Result<String> {
-    let stream = deploy
-        .base_metadata
-        .stream
-        .as_ref()
-        .ok_or_else(|| anyhow!("Missing `fedora-coreos.stream` in commit metadata"))?;
-    ensure!(!stream.is_empty(), "empty stream value");
-    Ok(stream.to_string())
+    if deploy.base_metadata.stream.is_none() {
+        fedora_coreos_stream_from_version(&deploy.version)
+    } else {
+        let stream = deploy
+            .base_metadata
+            .stream
+            .as_ref()
+            .ok_or_else(|| anyhow!("Missing `fedora-coreos.stream` in commit metadata"))?;
+        ensure!(!stream.is_empty(), "empty stream value");
+        Ok(stream.to_string())
+    }
+}
+
+fn fedora_coreos_stream_from_version(version: &String) -> Result<String> {
+    //41.20241109.3.0
+    let version_split: Vec<&str> = version.split('.').collect();
+    if version_split.len() != 4 {
+        return Err(anyhow!("{version} is not a valid CoreOS version ID"));
+    } else {
+        let stream_id: i32 = version_split[2]
+            .parse()
+            .map_err(|_| anyhow!("invalid Coreos version ID"))?;
+
+        match stream_id {
+            1 => Ok("next".to_string()),
+            2 => Ok("testing".to_string()),
+            3 => Ok("stable".to_string()),
+            _ => Err(anyhow!("This stream is not supported by Zincati")),
+        }
+    }
```

Comment:
This is a workaround for https://github.com/coreos/rpm-ostree/issues/5196
I tried to fix the parsing to properly handle the nested JSON but this proved difficult, so i went this route. 
This is a stopgap solution until we use `custom-origin-url` to fetch the update graph from an OCI artifact wanyway.